<b>Meta-analysis of epigenome-wide association studies in Alzheimer’s disease highlights 220 differentially methylated loci across cortex</b>

<i>Rebecca G. Smith, Ehsan Pishva, Gemma Shireby, Adam R. Smith, Janou A.Y. Roubroeks, Eilis Hannon, Gregory Wheildon, Diego Mastroeni, Gilles Gasparoni, Matthias Riemenschneider, Armin Giese, Andrew J. Sharp, Leonard Schalkwyk, Vahram Haroutunian, Wolfgang Viechtbauer, Daniel L.A. van den Hove, Michael Weedon, Jörn Walter, Paul D. Coleman, David A. Bennett, Philip L. De Jager, Jonathan Mill, Katie Lunnon</i>

Epigenome-wide association studies of Alzheimer’s disease have highlighted neuropathology-associated DNA methylation differences, although existing studies have been limited in sample size and utilized different brain regions. Here, we combine data from six methylomic studies of Alzheimer’s disease (N=1,453 unique individuals) to identify differential methylation associated with Braak stage in different brain regions and across cortex. At an experiment-wide significance threshold (P < 1.238 x 10<sup>−7</sup>) we identified 236 CpGs in the prefrontal cortex, 95 CpGs in the temporal gyrus and ten CpGs in the entorhinal cortex, with none in the cerebellum. Our cross-cortex meta-analysis (N=1,408 donors) identified 220 CpGs associated with neuropathology, annotated to 121 genes, of which 96 genes had not been previously reported at experiment-wide significance. Polyepigenic scores derived from these 220 CpGs explain 24.7% of neuropathological variance, whilst polygenic scores accounted for 20.2% of variance in these samples. The meta-analysis summary statistics are available in our online data resource (www.epigenomicslab.com/ad-meta-analysis/).

Paper uploaded to Biorxiv (https://www.biorxiv.org/content/10.1101/2020.02.28.957894v1)/
